integrationm/test/com/cultofbits/customizations/audit/AuditUtilsTest.groovy

```diff
@@ -0,0 +1,239 @@
+                        aFieldDefinition().id(14).name("RefField").description("")
+                                .setConfigurationKey("AutoRefField", "{\"args\":{\"source_field\":\"client\",\"field_name\":\"age\"}}")
+                )
+    }
+    def getDefinition(String name) {
+        return aDefinition()
+                .id(1)
+                .name(name)
+                .fieldDefinitions(
+                        aFieldDefinition().id(11).name("Text").description("\$text"),
+                        aFieldDefinition().id(12).name("Creator").description("\$audit.creator.username"),
+                        aFieldDefinition().id(13).name("TimeCreation").description("\$audit.updater.time"),
+                        aFieldDefinition().id(14).name("RefField").description("")
+                                .setConfigurationKey("AutoRefField", "{\"args\":{\"source_field\":\"client\",\"field_name\":\"age\"}}")
+                )
+    }
+
+    def "can detect \$audit fields on add"() {
+
+        def definition = getDefinition().build()
+        def newInstanceMsg = RecordmMsgBuilder.aMessage("cob-admin", definition, "add")
+                .build()
+
+        def auditFields = AuditInstanceUpdater.getAuditFields(null, newInstanceMsg, definition)
+
+        expect:
+        auditFields.size() == 2
+        auditFields[0].name == "Creator"
+        auditFields[1].name == "TimeCreation"
+    }
+
+    def "can detect \$audit fields on update with no audit arguments"() {
+
+        def definition = getDefinition().build()
+
+        //audit without arguments
+        def newInstanceMsg = RecordmMsgBuilder.aMessage("cob-admin", definition, "update").newField(definition.getFirstField("Text"), "text")
+                .build()
+
+        def auditFields = AuditInstanceUpdater.getAuditFields(null, newInstanceMsg, definition)
+
+        expect:
+        auditFields.size() == 2
+        auditFields[0].name == "Creator"
+        auditFields[1].name == "TimeCreation"
+
+    }
+
+    def "can detect \$audit fields on update with audit arguments [ignoreRefs:false,a:1] or [ignoreRefs:true,a:1]"() {
+
+        def definitionBuilder = getDefinition()
+
+        def definition = definitionBuilder.addFieldDefinitions(aFieldDefinition()
+                .id(14)
+                .name("TimeCreation2")
+                .description("\$audit.updater.uri([ignoreRefs:false,a:1])")
+                .setConfigurationExtensions("{\$audit.updater.uri", "{\"args\":[\"[ignoreRefs:false,a:1]\"]}}")
+                .build())
+                .addFieldDefinitions(aFieldDefinition()
+                        .id(16)
+                        .name("TimeCreation3")
+                        .description("\$audit.updater.username([ignoreRefs:true,a:1])")
+                        .setConfigurationExtensions("{\$audit.updater.username", "{\"args\":[\"[ignoreRefs:true,a:1]\"]}}")
+                        .build())
+                .build();
```

Comment:
I believe this would be easier to read:
```groovy
def definition = definitionBuilder
        .addFieldDefinitions(
                aFieldDefinition(14, "TimeCreation2", "\$audit.updater.uri([ignoreRefs:false,a:1])")
                        .setConfigurationExtensions("{\$audit.updater.uri", "{\"args\":[\"[ignoreRefs:false,a:1]\"]}}"))
        .addFieldDefinitions(
                aFieldDefinition(16, "TimeCreation3", "\$audit.updater.username([ignoreRefs:true,a:1])")
                        .setConfigurationExtensions("{\$audit.updater.username", "{\"args\":[\"[ignoreRefs:true,a:1]\"]}}")
        )
        .build();
```